### ISSUE TITLE
Still getting `XXX is referenced by YYY but not included in the documentation.`

### ISSUE THREAD

**Issue by oliviera9** [nlepage/typedoc-plugin-resolve-crossmodule-references#12]
Hello @nlepage ,
I tried to use your plugin for crosslinks between packages, but unfortunately it doesn't work as the usual warnings appear.
My environment is that:
```
    "typedoc": "^0.23.16",
    "typedoc-plugin-resolve-crossmodule-references": "^0.3.1",
    "typescript": "4.7.4"
```
and I am using
```shell
$ npx typedoc --plugin typedoc-plugin-resolve-crossmodule-references  --entryPointStrategy packages packages/*
```
to build the documentation.

Let me know if I can do further tests to understand where the problem is.

**Comment by nlepage** [comment]
Hi @oliviera9 
Can you tell me how is `YYY` using `XXX`? (e.g. `YYY` is a class that extends class `XXX`)
Is it OK for you to provide the output of typedoc execution?

**Comment by oliviera9** [comment]
@nlepage Actually there are 10 packages... but mainly, most of them implement an abstract class from a package `XXX`, then use constant values/enum defined in another package `YYY`.

Here is the output:
```
./packages/assets-algorand/src/algorand-provider.ts:7:0 - warning SignatureProvider is referenced by AlgorandAsset.BasicSignatureProvider but not included in the documentation.

7    export interface SignatureProvider {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by AlgorandAsset.AlgorandAsset._node but not included in the documentation.

36    declare class Node {

./packages/node/dist/index.d.ts:22:0 - warning AssetInfo is referenced by AlgorandAsset.AlgorandAssetBuilder._assetInfo but not included in the documentation.

22    declare type AssetInfo = {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by AlgorandAsset.AlgorandAssetBuilder._blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by AlgorandAsset.AlgorandAssetBuilder._chainId but not included in the documentation.

1    declare enum ChainId {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by AlgorandAsset.AlgorandAssetBuilder._network but not included in the documentation.

40    declare enum Network {

./packages/node/dist/index.d.ts:22:0 - warning AssetInfo is referenced by Entities.Asset._assetInfo but not included in the documentation.

22    declare type AssetInfo = {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by Entities.Asset._node but not included in the documentation.

36    declare class Node {

./packages/constants/dist/index.d.ts:17:0 - warning BlockchainType is referenced by Entities.Asset._type but not included in the documentation.

17    declare enum BlockchainType {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by Entities.AssetBuilder._blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by Entities.AssetBuilder._chainId but not included in the documentation.

1    declare enum ChainId {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by Entities.AssetBuilder._network but not included in the documentation.

40    declare enum Network {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by EosioAsset.EosioAsset._node but not included in the documentation.

36    declare class Node {

./packages/node/dist/index.d.ts:22:0 - warning AssetInfo is referenced by EosioAsset.EosioAssetBuilder._assetInfo but not included in the documentation.

22    declare type AssetInfo = {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by EosioAsset.EosioAssetBuilder._blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by EosioAsset.EosioAssetBuilder._chainId but not included in the documentation.

1    declare enum ChainId {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by EosioAsset.EosioAssetBuilder._network but not included in the documentation.

40    declare enum Network {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by EvmAsset.EvmAsset._node but not included in the documentation.

36    declare class Node {

./packages/node/dist/index.d.ts:22:0 - warning AssetInfo is referenced by EvmAsset.EvmAssetBuilder._assetInfo but not included in the documentation.

22    declare type AssetInfo = {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by EvmAsset.EvmAssetBuilder._blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by EvmAsset.EvmAssetBuilder._chainId but not included in the documentation.

1    declare enum ChainId {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by EvmAsset.EvmAssetBuilder._network but not included in the documentation.

40    declare enum Network {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by Helpers.validators.chainIdToAddressValidatorMap but not included in the documentation.

1    declare enum ChainId {

./packages/swap/src/swap.ts:7:0 - warning DestinationInfo is referenced by Swap.Swap._destinationAssets but not included in the documentation.

7    export type DestinationInfo = {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by Swap.Swap._node but not included in the documentation.

36    declare class Node {

./packages/entities/dist/index.d.ts:13:0 - warning Asset is referenced by Swap.Swap._sourceAsset but not included in the documentation.

13    declare abstract class Asset {

./packages/node/dist/index.d.ts:36:0 - warning Node is referenced by UtxoAsset.UtxoAsset._node but not included in the documentation.

36    declare class Node {

./packages/node/dist/index.d.ts:22:0 - warning AssetInfo is referenced by UtxoAsset.UtxoAssetBuilder._assetInfo but not included in the documentation.

22    declare type AssetInfo = {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by UtxoAsset.UtxoAssetBuilder._blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:1:0 - warning ChainId is referenced by UtxoAsset.UtxoAssetBuilder._chainId but not included in the documentation.

1    declare enum ChainId {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by UtxoAsset.UtxoAssetBuilder._network but not included in the documentation.

40    declare enum Network {

./packages/entities/dist/index.d.ts:13:0 - warning Asset is referenced by AlgorandAsset.AlgorandAssetBuilder.build.build but not included in the documentation.

13    declare abstract class Asset {

./packages/entities/dist/index.d.ts:13:0 - warning Asset is referenced by EosioAsset.EosioAssetBuilder.build.build but not included in the documentation.

13    declare abstract class Asset {

./packages/entities/dist/index.d.ts:13:0 - warning Asset is referenced by EvmAsset.EvmAssetBuilder.build.build but not included in the documentation.

13    declare abstract class Asset {

./packages/node/src/node.ts:30:0 - warning NativeDepositAddress is referenced by Node.Node.getNativeDepositAddress.getNativeDepositAddress but not included in the documentation.

30    export type NativeDepositAddress = {

./packages/node/src/node.ts:15:0 - warning TransactionStatus is referenced by Node.Node.getTransactionStatus.getTransactionStatus but not included in the documentation.

15    export type TransactionStatus = {

./packages/node/dist/index.d.ts:13:0 - warning InnerTransactionStatus is referenced by Swap.Swap.execute.execute but not included in the documentation.

13    declare type InnerTransactionStatus = {

./packages/assets-utxo/src/utxo-provider.ts:10:0 - warning Utxo is referenced by UtxoAsset.BlockstreamUtxoProvider.getUtxoByAddress.getUtxoByAddress but not included in the documentation.

10    export type Utxo = {

./packages/assets-utxo/src/utxo-provider.ts:5:0 - warning Transaction is referenced by UtxoAsset.BlockstreamUtxoProvider.waitForTransactionConfirmation.waitForTransactionConfirmation but not included in the documentation.

5    export type Transaction = {

./packages/entities/dist/index.d.ts:13:0 - warning Asset is referenced by UtxoAsset.UtxoAssetBuilder.build.build but not included in the documentation.

13    declare abstract class Asset {

./packages/assets-algorand/src/algorand-asset.ts:10:0 - warning AlgorandAssetConfig is referenced by AlgorandAsset.AlgorandAsset.constructor.new AlgorandAsset.config but not included in the documentation.

10    export type AlgorandAssetConfig = AssetConfig & {

./packages/assets-eosio/src/eosio-asset.ts:15:0 - warning EosioAssetConfig is referenced by EosioAsset.EosioAsset.constructor.new EosioAsset.config but not included in the documentation.

15    export type EosioAssetConfig = AssetConfig & {

./packages/assets-eosio/src/eosio-provider.ts:10:0 - warning Action is referenced by EosioAsset.EosioProvider.transact.transact.actions but not included in the documentation.

10    export type Action = {

./packages/assets-evm/src/evm-asset.ts:17:0 - warning EvmAssetConfig is referenced by EvmAsset.EvmAsset.constructor.new EvmAsset.config but not included in the documentation.

17    export type EvmAssetConfig = AssetConfig & { provider?: EvmProvider }

./packages/assets-evm/src/evm-provider.ts:16:0 - warning MakeContractCallOptions is referenced by EvmAsset.EvmProvider.makeContractCall.makeContractCall._options but not included in the documentation.

16    export type MakeContractCallOptions = {

./packages/assets-evm/src/evm-provider.ts:8:0 - warning MakeContractSendOptions is referenced by EvmAsset.EvmProvider.makeContractSend.makeContractSend._options but not included in the documentation.

8    export type MakeContractSendOptions = {

./packages/constants/dist/index.d.ts:24:0 - warning Blockchain is referenced by Helpers.maps.chainIdToBlockchainMap.__type.blockchain but not included in the documentation.

24    declare enum Blockchain {

./packages/constants/dist/index.d.ts:40:0 - warning Network is referenced by Helpers.maps.chainIdToBlockchainMap.__type.network but not included in the documentation.

40    declare enum Network {

./packages/providers/src/http-provider.ts:11:0 - warning CallTypes is referenced by Providers.HttpProvider.call.call._callType but not included in the documentation.

11    export enum CallTypes {

./packages/assets-utxo/src/utxo-provider.ts:17:0 - warning CallTypes is referenced by UtxoAsset.BlockstreamUtxoProvider._makeApiCall._makeApiCall._callType but not included in the documentation.

17    export enum CallTypes {

./packages/assets-utxo/src/utxo-asset.ts:12:0 - warning UtxoAssetConfig is referenced by UtxoAsset.UtxoAsset.constructor.new UtxoAsset.config but not included in the documentation.

12    export type UtxoAssetConfig = AssetConfig & { provider?: UtxoProvider }

Documentation generated at ./docs
```

What I notice is that the dist directory within every package is missing `*.d.ts.map`. That's because I use rollup to create a bundle and have a clean dist directory. Can it explain?

**Comment by nlepage** [comment]
> What I notice is that the dist directory within every package is missing `*.d.ts.map`. That's because I use rollup to create a bundle and have a clean dist directory. Can it explain?

`*.d.ts.map` files aren't used by the plugin.

The plugin matches the types by name and module name.

**Comment by oliviera9** [comment]
Do you see something strange from the output? Do you want to see other info?

**Comment by nlepage** [comment]
> Do you see something strange from the output? Do you want to see other info?

I can't see anything that would explain why it doesn't work in the output.
Is this opensource?

**Comment by oliviera9** [comment]
I am building a minimum working example that would simplify things.

**Comment by oliviera9** [comment]
@nlepage Here it is https://github.com/oliviera9/typedoc-crosslinks-fix

**Comment by oliviera9** [comment]
I think I understand the problem! in every `package.json` I set `displayName` to be equal to the package folder name. Now warnings have gone!

**Comment by nlepage** [comment]
> I think I understand the problem! in every `package.json` I set `displayName` to be equal to the package folder name. Now warnings have gone!

Good catch, I'll try to add a failing test for this.

**Comment by oliviera9** [comment]
I don't know if it makes sense, but you may apply a mapping from package directory name to `displayName` before resolving cross-references.

**Comment by nlepage** [comment]
@oliviera9 Just published v0.3.2 to fix this, let me know if this OK for you

**Comment by oliviera9** [comment]
It works perfectly! Thanks a lot for the quick support @nlepage
